optuna/samplers/_cmaes.py

```diff
@@ -393,6 +396,9 @@ def sample_independent(
         param_name: str,
         param_distribution: BaseDistribution,
     ) -> Any:
+
+        self._raise_error_if_multi_objective(study)
```

Comment:
Shouldn't cause trouble for now but I sensed some code smell. Just leaving comments if you have any thoughts on those. 

- These calls from `sample_independent` should never raise in practice since a user shouldn't directly call these sampler methods and `sample_relative` is always called prior to this
- `sample_relative` is not covered by the `catch` argument to `Study.optimize` while `sample_independent` is
- Introducing `BaseSampler.before_trial` might make sense
- Introducing a Optuna-defined error might be more user friendly

Reply:
Thanks for your insightful consideration. I understood your concern. Adding `BaseSampler.before_trial` seems reasonable to me, since we will introduce `BaseSampler.after_trial` in #2134.

However, I think we can merge this PR as a temporal fix. I'd like to implement `BaseSampler.before_trial` aligning with #2134 after this PR. What do you think?

Reply:
Sure, sounds good. I think we can clarify the second bullet by reconsidering the API in that way.